discordmd/discordmd.go

```diff
@@ -9,16 +9,45 @@ var (
 	italicRegex        = regexp.MustCompile(`(?ms)\*(.*?)\*`)
 	underlineRegex     = regexp.MustCompile(`(?ms)__(.*?)__`)
 	strikeThroughRegex = regexp.MustCompile(`(?ms)~~(.*?)~~`)
+	spoilerRegex       = regexp.MustCompile(`(?ms)\|\|(.*?)\|\|`)
 )
 
 // Parse parses Discord-flavored markdown to tview's [Color Tags].
 //
 // [Color Tags]: https://pkg.go.dev/github.com/rivo/tview#hdr-Colors
 func Parse(md string) string {
+	return ParseWithSpoilers(md, false)
+}
+
+// ParseWithSpoilers parses Discord-flavored markdown to tview’s [Color Tags]
+//
+// This modular version lets the caller choose to display spoilers as shown or hidden depending on the value of showSpoilers
+func ParseWithSpoilers(md string, showSpoilers bool) string {
+
 	md = boldRegex.ReplaceAllString(md, "[::b]$1[::-]")
 	md = italicRegex.ReplaceAllString(md, "[::i]$1[::-]")
 	md = underlineRegex.ReplaceAllString(md, "[::u]$1[::-]")
 	md = strikeThroughRegex.ReplaceAllString(md, "[::s]$1[::-]")
 
+	// parse spoiler tag
+	if showSpoilers {
+		md = spoilerRegex.ReplaceAllString(md, "[#E0E0E0:#383838:]$1[-:-:]")
```

Comment:
Aren't the colors supposed to be customized?

Reply:
Well, if discordo doesn't support custom colors by default.

Reply:
There are already some custom colors, for example in `ui/builder.go`, in the `buildMentions` function
```Go
if mUser.ID == clientID {
	color = "[:#5865F2]"
} else {
	color = "[#EB459E]"
}
```